### ISSUE TITLE
swagger-codegen not installed in latest docker

### ISSUE THREAD

**Issue by danielbons** [swagger-api/swagger-codegen#12321]
##### Description
Is seems like swagger-codegen is not installed in the latest release docker, 3.0.52. Trying to generate output from an API specification gives the following error: 

`docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "generate": executable file not found in $PATH: unknown.`

My interpretation is that swagger-codegen should be installed on /opt in the docker images but that path is empty for 3.0.52

##### Swagger-codegen version

Using latest released docker image 3.0.52

##### Swagger declaration file content or url

N/A

##### Command line used for generation

```
$ docker run --rm -v /local/home/maxarcat/:/maxarcat swaggerapi/swagger-codegen-cli-v3:3.0.52 generate \
   --input-spec /maxarcat/discovery-api.json \
   --lang python \
   --output /maxarcat/maxarcat_client \
   --reserved-words-mappings datetime=search_datetime \
   --config /maxarcat/maxarcat_client_config.json
```

##### Steps to reproduce

`docker run --rm swaggerapi/swagger-codegen-cli-v3:3.0.52 generate`

Any call to `generate` results in the error: 

docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "generate": executable file not found in $PATH: unknown.

Verifying that opt is empty can be performed by:
```
$ docker run -it --rm swaggerapi/swagger-codegen-cli-v3:3.0.52 sh
/generator # ls /opt
```

**Comment by violen** [comment]
Same here. Last working Docker Image of the v3 CLI was 3.0.46.
Also it is an amd64 only image.

A fix for the newly multiarch build process by @frantuma would be awesome

**Comment by qhaas** [comment]
If I'm following [the CI script](https://github.com/swagger-api/swagger-codegen/blob/v3.0.52/.github/workflows/docker-release-3.0.yml#L87) correctly, it uses [the minimal dockerfile](https://github.com/swagger-api/swagger-codegen/blob/v3.0.52/modules/swagger-generator/Dockerfile_minimal) instead of the [swagger codegen cli dockerfile](https://github.com/swagger-api/swagger-codegen/blob/v3.0.52/modules/swagger-codegen-cli/Dockerfile)

Which would explain why `swaggerapi/swagger-generator-v3-minimal:3.0.52` and `swaggerapi/swagger-codegen-cli-v3:3.0.52` have the same hash

**Comment by NicolasCARPi** [comment]
Hello everyone. It would be marvelous if this could be adressed in priority. I'd argue an issue that prevents completely the execution of the main program should be top priority, shouldn't it? As @qhaas mentioned, it seems the Dockerfiles got mixed up and the wrong image has been built.

Thanks for all the work everyone! :)

Cheers,
~Nicolas

**Comment by tariqajyusuf** [comment]
In the meantime, while we wait for this PR to get looked at. I got a temporary workaround by using `eclipse-temurin` and just running the [CLI download steps](https://github.com/swagger-api/swagger-codegen?tab=readme-ov-file#prerequisites)

**Comment by frantuma** [comment]
Thanks all for reporting, investigating and providing the fix. Main issue has been fixed in #12331 and is available in 3.0.54 release. Swagger Generator "minimal" image is still failing and will be addressed in different ticket scope
